src/doc/unstable-book/src/compiler-flags/llvm-module-flag.md

```diff
@@ -0,0 +1,12 @@
+# `llvm-module-flag`
+
+---------------------
+
+This flag allows adding a key/value to the `!llvm.module.flags` metadata in the
+LLVM-IR for a compiled Rust module.  The syntax is
+
+`-Z llvm_module_flag=<name>:<type>:<value>:<behavior>`
+
+Currently only u32 values are supported but the type is required to be specified
+for forward compatibility.  The `behavior` element must match one of the named
+LLVM [metadata behaviors](https://llvm.org/docs/LangRef.html#module-flags-metadata)
```

Comment:
It might be good to have a practical example of why you might want to use this flag here.

Reply:
For setting the small data limit to 0 to prevent the use of .sdata (the purpose of this PR), the flag can be specified as `-Zllvm_module_flag=SmallDataLimit:u32:0:Error`